Flaky TestHiveMerge.testMergeSimpleSelect and TestHiveMerge.testMergeSimpleSelect

```
2022-10-19T12:56:24.8621981Z tests               | 2022-10-19 18:41:24 INFO: FAILURE     /    io.trino.tests.product.hive.TestHiveMerge.testMergeSimpleSelect (Groups: hive_transactional) took 5.2 seconds
2022-10-19T12:56:24.8623168Z tests               | 2022-10-19 18:41:24 SEVERE: Failure cause:
2022-10-19T12:56:24.8624111Z tests               | java.lang.AssertionError: [onTrino] Expected row count to be <4>, but was <5>; rows=[[Ed, 14, Etherville], [Dave, 33, Darbyshire], [Aaron, 17, Arches], [Carol, 9, Centreville], [Bill, 7, Buena]]
```

```
2022-10-19T12:56:16.2185695Z tests               | 2022-10-19 18:41:16 INFO: FAILURE     /    io.trino.tests.product.hive.TestHiveMerge.testMergeSimpleQuery (Groups: hive_transactional) took 5.0 seconds
2022-10-19T12:56:16.2186686Z tests               | 2022-10-19 18:41:16 SEVERE: Failure cause:
2022-10-19T12:56:16.2187484Z tests               | java.lang.AssertionError: [onTrino] Expected row count to be <4>, but was <5>; rows=[[Aaron, 17, Arches], [Dave, 33, Darbyshire], [Carol, 9, Centreville], [Ed, 14, Etherville], [Bill, 7, Buena]]
```

https://github.com/trinodb/trino/actions/runs/3281446960/jobs/5403713341

Before the SELECT is run, the MERGE had already failed with a backtrace showing a collision writing the `_orc_acid_version` file in the `delta_0000002_0000002_0000` directory.  I'll look into ways that could happen.

FTR, ISTM that exception should have propagated out and failed the test before the SELECT was attempted.

```
2022-10-19T12:56:11.2158400Z tests               | 2022-10-19 18:41:11 INFO: [178 of 553] io.trino.tests.product.hive.TestHiveMerge.testMergeSimpleQuery (Groups: hive_transactional)
2022-10-19T12:56:11.7139893Z presto-master       | 2022-10-19T18:41:11.713+0545	INFO	dispatcher-query-70	io.trino.event.QueryMonitor	TIMELINE: Query 20221019_125611_01016_2b4dm :: FINISHED :: elapsed 40ms :: planning 0ms :: waiting 0ms :: scheduling 40ms :: running 0ms :: finishing 40ms :: begin 2022-10-19T18:41:11.673+05:45 :: end 2022-10-19T18:41:11.713+05:45
2022-10-19T12:56:12.6290281Z presto-master       | 2022-10-19T18:41:12.628+0545	INFO	dispatcher-query-97	io.trino.event.QueryMonitor	TIMELINE: Query 20221019_125611_01017_2b4dm :: FINISHED :: elapsed 912ms :: planning 14ms :: waiting 8ms :: scheduling 14ms :: running 3ms :: finishing 881ms :: begin 2022-10-19T18:41:11.716+05:45 :: end 2022-10-19T18:41:12.628+05:45
2022-10-19T12:56:13.6058261Z presto-master       | 2022-10-19T18:41:13.604+0545	ERROR	SplitRunner-6-97	io.trino.execution.executor.TaskExecutor	Error processing Split 20221019_125612_01018_2b4dm.0.0.0-2  (start = 708311.339701, wall = 872 ms, cpu = 0 ms, wait = 0 ms, calls = 3): HIVE_FILESYSTEM_ERROR: Exception writing _orc_acid_version file for deltaDirectory hdfs://hadoop-master:9000/user/hive/warehouse/test_merge_simple_target_false_none_1pqiqlc2ma5a/delta_0000002_0000002_0000
2022-10-19T12:56:13.6059753Z presto-master       | io.trino.spi.TrinoException: Exception writing _orc_acid_version file for deltaDirectory hdfs://hadoop-master:9000/user/hive/warehouse/test_merge_simple_target_false_none_1pqiqlc2ma5a/delta_0000002_0000002_0000
2022-10-19T12:56:13.6060747Z presto-master       | 	at io.trino.plugin.hive.HiveMetadata.createOrcAcidVersionFile(HiveMetadata.java:2139)
2022-10-19T12:56:13.6061561Z presto-master       | 	at io.trino.plugin.hive.HiveMetadata.lambda$finishMerge$53(HiveMetadata.java:1859)
2022-10-19T12:56:13.6062329Z presto-master       | 	at java.base/java.util.Optional.ifPresent(Optional.java:178)
2022-10-19T12:56:13.6063016Z presto-master       | 	at io.trino.plugin.hive.HiveMetadata.finishMerge(HiveMetadata.java:1859)
2022-10-19T12:56:13.6063874Z presto-master       | 	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.finishMerge(ClassLoaderSafeConnectorMetadata.java:1070)
2022-10-19T12:56:13.6064551Z presto-master       | 	at io.trino.metadata.MetadataManager.finishMerge(MetadataManager.java:1071)
2022-10-19T12:56:13.6065080Z presto-master       | 	at io.trino.sql.planner.LocalExecutionPlanner.lambda$createTableFinisher$4(LocalExecutionPlanner.java:4065)
2022-10-19T12:56:13.6065622Z presto-master       | 	at io.trino.operator.TableFinishOperator.getOutput(TableFinishOperator.java:319)
2022-10-19T12:56:13.6066118Z presto-master       | 	at io.trino.operator.Driver.processInternal(Driver.java:411)
2022-10-19T12:56:13.6066499Z presto-master       | 	at io.trino.operator.Driver.lambda$process$10(Driver.java:314)
2022-10-19T12:56:13.6066943Z presto-master       | 	at io.trino.operator.Driver.tryWithLock(Driver.java:706)
2022-10-19T12:56:13.6067312Z presto-master       | 	at io.trino.operator.Driver.process(Driver.java:306)
2022-10-19T12:56:13.6067990Z presto-master       | 	at io.trino.operator.Driver.processForDuration(Driver.java:277)
2022-10-19T12:56:13.6068578Z presto-master       | 	at io.trino.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:736)
2022-10-19T12:56:13.6069451Z presto-master       | 	at io.trino.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:164)
2022-10-19T12:56:13.6070430Z presto-master       | 	at io.trino.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:515)
2022-10-19T12:56:13.6071509Z presto-master       | 	at io.trino.$gen.Trino_400_68_gd124276____20221019_124833_2.run(Unknown Source)
2022-10-19T12:56:13.6072487Z presto-master       | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-10-19T12:56:13.6073426Z presto-master       | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-10-19T12:56:13.6074254Z presto-master       | 	at java.base/java.lang.Thread.run(Thread.java:833)
2022-10-19T12:56:13.6075672Z presto-master       | Caused by: org.apache.hadoop.ipc.RemoteException(java.io.IOException): File /user/hive/warehouse/test_merge_simple_target_false_none_1pqiqlc2ma5a/delta_0000002_0000002_0000/_orc_acid_version could only be written to 0 of the 1 minReplication nodes. There are 1 datanode(s) running and no node(s) are excluded in this operation.
2022-10-19T12:56:13.6077132Z presto-master       | 	at org.apache.hadoop.hdfs.server.blockmanagement.BlockManager.chooseTarget4NewBlock(BlockManager.java:2121)
2022-10-19T12:56:13.6078355Z presto-master       | 	at org.apache.hadoop.hdfs.server.namenode.FSDirWriteFileOp.chooseTargetForNewBlock(FSDirWriteFileOp.java:286)
2022-10-19T12:56:13.6079674Z presto-master       | 	at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.getAdditionalBlock(FSNamesystem.java:2706)
2022-10-19T12:56:13.6080743Z presto-master       | 	at org.apache.hadoop.hdfs.server.namenode.NameNodeRpcServer.addBlock(NameNodeRpcServer.java:875)
2022-10-19T12:56:13.6082008Z presto-master       | 	at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolServerSideTranslatorPB.addBlock(ClientNamenodeProtocolServerSideTranslatorPB.java:561)
2022-10-19T12:56:13.6083622Z presto-master       | 	at org.apache.hadoop.hdfs.protocol.proto.ClientNamenodeProtocolProtos$ClientNamenodeProtocol$2.callBlockingMethod(ClientNamenodeProtocolProtos.java)
2022-10-19T12:56:13.6084692Z presto-master       | 	at org.apache.hadoop.ipc.ProtobufRpcEngine$Server$ProtoBufRpcInvoker.call(ProtobufRpcEngine.java:524)
2022-10-19T12:56:13.6085459Z presto-master       | 	at org.apache.hadoop.ipc.RPC$Server.call(RPC.java:1025)
2022-10-19T12:56:13.6086177Z presto-master       | 	at org.apache.hadoop.ipc.Server$RpcCall.run(Server.java:876)
2022-10-19T12:56:13.6087069Z presto-master       | 	at org.apache.hadoop.ipc.Server$RpcCall.run(Server.java:822)
2022-10-19T12:56:13.6087806Z presto-master       | 	at java.security.AccessController.doPrivileged(Native Method)
2022-10-19T12:56:13.6088516Z presto-master       | 	at javax.security.auth.Subject.doAs(Subject.java:422)
2022-10-19T12:56:13.6089341Z presto-master       | 	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1730)
2022-10-19T12:56:13.6090101Z presto-master       | 	at org.apache.hadoop.ipc.Server$Handler.run(Server.java:2682)
2022-10-19T12:56:13.6090682Z presto-master       | 
2022-10-19T12:56:13.6091318Z presto-master       | 	at org.apache.hadoop.ipc.Client.getRpcResponse(Client.java:1511)
2022-10-19T12:56:13.6092078Z presto-master       | 	at org.apache.hadoop.ipc.Client.call(Client.java:1457)
2022-10-19T12:56:13.6092728Z presto-master       | 	at org.apache.hadoop.ipc.Client.call(Client.java:1367)
2022-10-19T12:56:13.6093584Z presto-master       | 	at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:228)
2022-10-19T12:56:13.6094493Z presto-master       | 	at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:116)
2022-10-19T12:56:13.6095262Z presto-master       | 	at jdk.proxy7/jdk.proxy7.$Proxy436.addBlock(Unknown Source)
2022-10-19T12:56:13.6096284Z presto-master       | 	at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.addBlock(ClientNamenodeProtocolTranslatorPB.java:513)
2022-10-19T12:56:13.6097306Z presto-master       | 	at jdk.internal.reflect.GeneratedMethodAccessor1213.invoke(Unknown Source)
2022-10-19T12:56:13.6098321Z presto-master       | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-10-19T12:56:13.6099300Z presto-master       | 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2022-10-19T12:56:13.6100211Z presto-master       | 	at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:422)
2022-10-19T12:56:13.6101219Z presto-master       | 	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeMethod(RetryInvocationHandler.java:165)
2022-10-19T12:56:13.6102210Z presto-master       | 	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invoke(RetryInvocationHandler.java:157)
2022-10-19T12:56:13.6103131Z presto-master       | 	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeOnce(RetryInvocationHandler.java:95)
2022-10-19T12:56:13.6104071Z presto-master       | 	at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:359)
2022-10-19T12:56:13.6104856Z presto-master       | 	at jdk.proxy7/jdk.proxy7.$Proxy437.addBlock(Unknown Source)
2022-10-19T12:56:13.6105624Z presto-master       | 	at org.apache.hadoop.hdfs.DFSOutputStream.addBlock(DFSOutputStream.java:1081)
2022-10-19T12:56:13.6106531Z presto-master       | 	at org.apache.hadoop.hdfs.DataStreamer.locateFollowingBlock(DataStreamer.java:1865)
2022-10-19T12:56:13.6107562Z presto-master       | 	at org.apache.hadoop.hdfs.DataStreamer.nextBlockOutputStream(DataStreamer.java:1668)
2022-10-19T12:56:13.6108456Z presto-master       | 	at org.apache.hadoop.hdfs.DataStreamer.run(DataStreamer.java:716)
```

--- a/testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveMerge.java
+++ b/testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveMerge.java
@@ -46,12 +46,12 @@ public class TestHiveMerge
     @Test(groups = HIVE_TRANSACTIONAL, timeOut = 60 * 60 * 1000)
     public void testMergeSimpleSelect()
     {
-        withTemporaryTable("merge_simple_target", true, false, NONE, targetTable -> {
+        withTemporaryTable("merge_simple_select_target", true, false, NONE, targetTable -> {
             onTrino().executeQuery(format("CREATE TABLE %s (customer VARCHAR, purchases INT, address VARCHAR) WITH (transactional = true)", targetTable));
 
             onTrino().executeQuery(format("INSERT INTO %s (customer, purchases, address) VALUES ('Aaron', 5, 'Antioch'), ('Bill', 7, 'Buena'), ('Carol', 3, 'Cambridge'), ('Dave', 11, 'Devon')", targetTable));
 
-            withTemporaryTable("merge_simple_source", true, false, NONE, sourceTable -> {
+            withTemporaryTable("merge_simple_select_source", true, false, NONE, sourceTable -> {
                 onTrino().executeQuery(format("CREATE TABLE %s (customer VARCHAR, purchases INT, address VARCHAR) WITH (transactional = true)", sourceTable));
 
                 onTrino().executeQuery(format("INSERT INTO %s (customer, purchases, address) VALUES ('Aaron', 6, 'Arches'), ('Ed', 7, 'Etherville'), ('Carol', 9, 'Centreville'), ('Dave', 11, 'Darbyshire')", sourceTable));
@@ -71,12 +71,12 @@ public class TestHiveMerge
     @Test(groups = HIVE_TRANSACTIONAL, timeOut = 60 * 60 * 1000)
     public void testMergeSimpleSelectPartitioned()
     {
-        withTemporaryTable("merge_simple_target", true, true, NONE, targetTable -> {
+        withTemporaryTable("merge_simple_select_partitioned_target", true, true, NONE, targetTable -> {
             onTrino().executeQuery(format("CREATE TABLE %s (customer VARCHAR, purchases INT, address VARCHAR) WITH (transactional = true, partitioned_by = ARRAY['address'])", targetTable));
 
             onTrino().executeQuery(format("INSERT INTO %s (customer, purchases, address) VALUES ('Aaron', 5, 'Antioch'), ('Bill', 7, 'Buena'), ('Carol', 3, 'Cambridge'), ('Dave', 11, 'Devon')", targetTable));
 
-            withTemporaryTable("merge_simple_source", true, false, NONE, sourceTable -> {
+            withTemporaryTable("merge_simple_select_partitioned_source", true, false, NONE, sourceTable -> {
                 onTrino().executeQuery(format("CREATE TABLE %s (customer VARCHAR, purchases INT, address VARCHAR) WITH (transactional = true)", sourceTable));
 
                 onTrino().executeQuery(format("INSERT INTO %s (customer, purchases, address) VALUES ('Aaron', 6, 'Arches'), ('Ed', 7, 'Etherville'), ('Carol', 9, 'Centreville'), ('Dave', 11, 'Darbyshire')", sourceTable));
@@ -97,7 +97,7 @@ public class TestHiveMerge
     public void testMergeUpdateWithVariousLayouts(boolean partitioned, String bucketing)
     {
         BucketingType bucketingType = bucketing.isEmpty() ? NONE : BUCKETED_V2;
-        withTemporaryTable("merge_with_various_formats", true, partitioned, bucketingType, targetTable -> {
+        withTemporaryTable("merge_update_with_various_formats", true, partitioned, bucketingType, targetTable -> {
             StringBuilder builder = new StringBuilder();
             builder.append("CREATE TABLE ")
                     .append(targetTable)
@@ -113,7 +113,7 @@ public class TestHiveMerge
             onTrino().executeQuery(format("INSERT INTO %s (customer, purchase) VALUES ('Dave', 'dates'), ('Lou', 'limes'), ('Carol', 'candles')", targetTable));
             verifySelectForTrinoAndHive("SELECT * FROM " + targetTable, "TRUE", row("Dave", "dates"), row("Lou", "limes"), row("Carol", "candles"));
 
-            withTemporaryTable("merge_simple_source", true, false, NONE, sourceTable -> {
+            withTemporaryTable("merge_update_with_various_formats_source", true, false, NONE, sourceTable -> {
                 onTrino().executeQuery(format("CREATE TABLE %s (customer VARCHAR, purchase VARCHAR) WITH (transactional = true)", sourceTable));
 
                 onTrino().executeQuery(format("INSERT INTO %s (customer, purchase) VALUES ('Craig', 'candles'), ('Len', 'limes'), ('Joe', 'jellybeans')", sourceTable));
@@ -133,13 +133,13 @@ public class TestHiveMerge
     @Test(groups = HIVE_TRANSACTIONAL, timeOut = TEST_TIMEOUT)
     public void testMergeUnBucketedUnPartitionedFailure()
     {
-        withTemporaryTable("merge_with_various_formats", true, false, NONE, targetTable -> {
+        withTemporaryTable("merge_with_various_formats_failure", true, false, NONE, targetTable -> {
             onTrino().executeQuery(format("CREATE TABLE %s (customer VARCHAR, purchase VARCHAR) WITH (transactional = true)", targetTable));
 
             onTrino().executeQuery(format("INSERT INTO %s (customer, purchase) VALUES ('Dave', 'dates'), ('Lou', 'limes'), ('Carol', 'candles')", targetTable));
             verifySelectForTrinoAndHive("SELECT * FROM " + targetTable, "TRUE", row("Dave", "dates"), row("Lou", "limes"), row("Carol", "candles"));
 
-            withTemporaryTable("merge_simple_source", true, false, NONE, sourceTable -> {
+            withTemporaryTable("merge_with_various_formats_failure_source", true, false, NONE, sourceTable -> {
                 onTrino().executeQuery(format("CREATE TABLE %s (customer VARCHAR, purchase VARCHAR) WITH (transactional = true)", sourceTable));
 
                 onTrino().executeQuery(format("INSERT INTO %s (customer, purchase) VALUES ('Craig', 'candles'), ('Len', 'limes'), ('Joe', 'jellybeans')", sourceTable));
@@ -263,7 +263,7 @@ public class TestHiveMerge
     @Test(groups = HIVE_TRANSACTIONAL, timeOut = 60 * 60 * 1000)
     public void testMergeSimpleQuery()
     {
-        withTemporaryTable("merge_simple_target", true, false, NONE, targetTable -> {
+        withTemporaryTable("merge_simple_query_target", true, false, NONE, targetTable -> {
             onTrino().executeQuery(format("CREATE TABLE %s (customer VARCHAR, purchases INT, address VARCHAR) WITH (transactional = true)", targetTable));
 
             onTrino().executeQuery(format("INSERT INTO %s (customer, purchases, address) VALUES ('Aaron', 5, 'Antioch'), ('Bill', 7, 'Buena'), ('Carol', 3, 'Cambridge'), ('Dave', 11, 'Devon')", targetTable));
@@ -301,7 +301,7 @@ public class TestHiveMerge
     @Test(groups = HIVE_TRANSACTIONAL, timeOut = 60 * 60 * 1000)
     public void testMergeSimpleQueryPartitioned()
     {
-        withTemporaryTable("merge_simple_target", true, true, NONE, targetTable -> {
+        withTemporaryTable("merge_simple_query_partitioned_target", true, true, NONE, targetTable -> {
             onTrino().executeQuery(format("CREATE TABLE %s (customer VARCHAR, purchases INT, address VARCHAR) WITH (transactional = true, partitioned_by = ARRAY['address'])", targetTable));
 
             onTrino().executeQuery(format("INSERT INTO %s (customer, purchases, address) VALUES ('Aaron', 5, 'Antioch'), ('Bill', 7, 'Buena'), ('Carol', 3, 'Cambridge'), ('Dave', 11, 'Devon')", targetTable));

